segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,230 @@
+        """
+        Returns a list of edge points. Computes edges using binary erosion and
+        exclusive or operation. Generates edges from a thresholded binary mask.
+        Args:
+            threshold: a float to threshold all values against to create binary
+                       mask
+            optional_array: an optional array to overload the function and
+            create edge points from different mask
+        Returns:
+            Contours: a list of coordinates that contain edge points
+        """
+        if optional_array is None:
+            optional_array = self.generate_binary_mask_from_threshold(threshold)
+        contours, hierarchy = cv2.findContours(np.uint8(optional_array),
+                                               cv2.RETR_TREE,
+                                               cv2.CHAIN_APPROX_NONE)
+        return np.squeeze(contours[0], axis=1)
+
+    def get_edge_mask(self, threshold: float,
+                      optional_array: np.array = None) -> np.array:
+        """
+        Returns a mask of edge points where edges of an roi are represented
+        as 1 and non edges are 0. Computes edges using binary erosion and
+        exclusive or operation. Generates edges from a thresholded binary mask.
+        Args:
+            threshold: a float to threshold all values against to create binary
+                       mask
+            optional_array: an optional array to overload the function and
+            create edge points from different mask
+
+        Returns:
+            mask_matrix: a binary mask of the edges given a threshold
+
+        """
+        edge_coordinates = self.get_edge_points(threshold=threshold,
+                                                optional_array=optional_array)
+        mask_matrix = np.zeros(shape=self.image_shape)
+        for edge_coordinate in edge_coordinates:
+            mask_matrix[edge_coordinate[0]][edge_coordinate[1]] = 1
+        return mask_matrix
+
+    def create_manifest_json(self, source_ref: str,
+                             video_source_ref: str, max_source_ref: str,
+                             avg_source_ref: str, trace_source_ref: str,
+                             roi_data_source_ref: str) -> str:
+        """
+        Function to make the manifest json string required for sagemaker
+        ground truth to read the data in successfully. The strings are
+        S3 bucket URIs for each of the required data.
+        Args:
+            source_ref: Location of ROI mask png
+            video_source_ref: Location of 2P video for the ROI
+            max_source_ref: Location of the maximum projection for the ROI
+            avg_source_ref: Location of the average projection for the ROI
+            trace_source_ref: Location of the trace data for the ROI
+            roi_data_source_ref: Location of the ROI coordinate data
+
+        Returns:
+            dictionary: returns dictionary as binary json string
```

Comment:
not sure about `binary`? Docs for `json.dumps` say `Serialize obj to a JSON formatted str`